server/src/services/media.service.ts

```diff
@@ -185,8 +195,18 @@ export class MediaService extends BaseService {
       pathsToDelete.push(thumbnailFile.path);
     }
 
+    if (extractedFile && extractedFile.path !== generated.extractedPath) {
+      this.logger.debug(`Deleting old extracted image for asset ${asset.id}`);
+      pathsToDelete.push(extractedFile.path);
+    }
+
     if (pathsToDelete.length > 0) {
-      await Promise.all(pathsToDelete.map((path) => this.storageRepository.unlink(path)));
+      await Promise.all(
+        pathsToDelete.map(async (path) => {
+          await this.storageRepository.unlink(path);
+          await this.assetRepository.removeAssetFile(path);
```

Comment:
This should only be called for the extracted path. It's unnecessary for the others. The signature should also be changed to be based on the asset ID and file type instead of the path.

Reply:
If the file doesn't exist then there shouldn't be an entry in the DB for it imo

Reply:
No special handling of `extractedFile` then it seems